File: python/chapter1/exercise1_test.py
import exercise1
import unittest

class Exercise1Test(unittest.TestCase):

    def setUp(self):
        self.input_data = ""
        self.expected_output = ""
        
    def _check_function(self, func):
        self.assertEqual(func(self.input_data), self.expected_output)

    def test_1_1(self):
        self.input_data = "hola"
        self.expected_output = True
        self._check_function(exercise1.run1)

    def test_1_2(self):
        self.input_data = "holo"
        self.expected_output = False
        self._check_function(exercise1.run2)

    def test_2_1(self):
        self.input_data = "hola"
        self.expected_output = True
        self._check_function(exercise1.run2)

    def test_2_2(self):
        self.input_data = "holo"
        self.expected_output = False
        self._check_function(exercise1.run2)

    def test_3_1(self):
        self.input_data = "hola"
        self.expected_output = True
        self._check_function(exercise1.run3)

    def test_3_2(self):
        self.input_data = "holo"
        self.expected_output = False
        self._check_function(exercise1.run3)

File: java/chapter1/Exercise1Test.java
import static org.junit.Assert.assertEquals;
import org.junit.Test;
import org.junit.Before;

public class Exercise1Test { 

    private String input_data;
    private boolean expected_output;
    private Exercise1 exercise1;

    @Before
    public void setUp() {
        this.exercise1 = new Exercise1();
    }

    @Test
    public void test1_1()  {
        input_data = "hola";
        expected_output = true;
        assertEquals(exercise1.uniqueCharacters(this.input_data), this.expected_output);
    }

    @Test
    public void test1_2()  {
        input_data = "holo";
        expected_output = false;
        assertEquals(exercise1.uniqueCharacters(this.input_data), this.expected_output);
    }

    @Test
    public void test2_1()  {
        input_data = "hola";
        expected_output = true;
        assertEquals(exercise1.uniqueCharacters2(this.input_data), this.expected_output);
    }

    @Test
    public void test2_2()  {
        input_data = "holo";
        expected_output = false;
        assertEquals(exercise1.uniqueCharacters2(this.input_data), this.expected_output);
    }
}
